tests/Vaults/daily-oracles.spec.ts

```diff
@@ -0,0 +1,60 @@
+import { dailyMetricsResponse } from './mocks';
+import { extractDailyOracles } from '@/utils';
+
+describe('Tests for Daily Oracles', () => {
+  it('should return an empty object when dailyMetricsResponse is undefined or null', () => {
+    const tokenName = 'ATOM';
+    let dailyMetricsResponse = undefined;
+
+    let result = extractDailyOracles(tokenName, dailyMetricsResponse);
+    expect(result).toEqual({});
+    dailyMetricsResponse = null;
+    result = extractDailyOracles(tokenName, dailyMetricsResponse);
+    expect(result).toEqual({});
+  });
+
+  it('should return an empty object when dailyMetricsResponse has empty nodes array', () => {
+    const tokenName = 'ATOM';
+    const dailyMetricsResponse = {
+      ATOM_oracle: {
+        nodes: [],
+      },
+    };
+
+    const result = extractDailyOracles(tokenName, dailyMetricsResponse);
+
+    expect(result).toEqual({});
+  });
+
```

Comment:
a more common scenarion would be :
```
  const dailyMetricsResponse = {
      ATOM_oracle: undefined,
    };
```
lets test for that as well

Reply:
Adding in #56